lib/Dialect/HW/HWTypes.cpp

```diff
@@ -261,6 +261,17 @@ Type StructType::getFieldType(mlir::StringRef fieldName) {
   return Type();
 }
 
+Optional<unsigned> StructType::getFieldIndex(mlir::StringRef fieldName) {
+  ArrayRef<hw::StructType::FieldInfo> elems = getElements();
+  unsigned idx = 0, numElems = elems.size();
+  for (; idx < numElems; ++idx)
+    if (elems[idx].name == fieldName)
+      break;
```

Comment:
nit: return idx here.

Reply:
Then the normal loop exit doesn't need the if either.

Reply:
Yeah, I copied this code from the call site and neglected to clean it up.